arrows/zlib/tests/test_bytestream_compressor.cxx

```diff
@@ -0,0 +1,213 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// Test bytestream_compressor class.
+
+#include <gtest/gtest.h>
+
+#include <arrows/zlib/bytestream_compressor.h>
+
+#include <array>
+#include <random>
+
+using namespace kwiver::vital;
+
+// ----------------------------------------------------------------------------
+int
+main( int argc, char** argv )
+{
+  ::testing::InitGoogleTest( &argc, argv );
+  return RUN_ALL_TESTS();
+}
+
+// ----------------------------------------------------------------------------
+class test_bytestream_compressor : public ::testing::Test
+{
+private:
+
+  void
+  SetUp() override
+  {
+    text_data.resize( data_size );
+    binary_data.resize( data_size );
+
+    // Create some repeating, compressable data
+    for( size_t i = 0; i < data_size; ++i )
+    {
+      text_data[ i ] = 'A' + ( i % 16 ) + i / ( 1 << 14 );
+      binary_data[ i ] = i % 64 + i / ( 1 << 12 );
+    }
+  }
+
+public:
+  constexpr static size_t data_size = 1 << 16; // 64 KB
+  std::vector< uint8_t > text_data;
+  std::vector< uint8_t > binary_data;
+};
+
+// ----------------------------------------------------------------------------
+TEST_F ( test_bytestream_compressor, round_trip_deflate_text )
+{
+  bytestream_compressor compressor(
+    bytestream_compressor::MODE_COMPRESS,
+    bytestream_compressor::COMPRESSION_TYPE_DEFLATE,
+    bytestream_compressor::DATA_TYPE_TEXT );
+
+  compressor.write( text_data );
+  compressor.flush();
+  auto const compressed_data = compressor.read();
+
+  EXPECT_EQ( 175, compressed_data.size() );
```

Comment:
Maybe define this expected value as a constexpr in case the size of the test data needs to change.